coral-presto/src/main/java/com/linkedin/coral/presto/parser/ParseTreeBuilder.java

```diff
@@ -0,0 +1,1500 @@
+      JoinCriteria joinCriteria = node.getCriteria().get();
+      natural = createLiteralBoolean(joinCriteria instanceof NaturalJoin, ZERO);
+      if (joinCriteria instanceof JoinOn) {
+        conditionType = JoinConditionType.ON;
+        condition = visitNode(((JoinOn) joinCriteria).getExpression(), context);
+      } else if (joinCriteria instanceof JoinUsing) {
+        conditionType = JoinConditionType.USING;
+        condition =
+            createSqlNodeList(getListSqlNode(((JoinUsing) joinCriteria).getColumns(), context), getParserPos(node));
+      }
+    }
+    JoinType joinType = JOIN_TYPE_MAP.get(node.getType());
+    if (joinType == null) {
+      throw getUnsupportedException(node);
+    }
+    return new SqlJoin(getParserPos(node), left, natural, joinType.symbol(ZERO), right, conditionType.symbol(ZERO),
+        condition);
+  }
+
+  @Override
+  protected SqlNode visitExists(ExistsPredicate node, ParserVisitorContext context) {
+    return EXISTS.createCall(getParserPos(node), visitNode(node.getSubquery(), context));
+  }
+
+  private RelDataType getDecimalType(String type) {
+    String value = type.substring(8, type.length() - 1);
+    String[] nums = value.split(",");
+    int precision = Integer.parseInt(nums[0]);
+    int scale = nums.length > 1 ? Integer.parseInt(nums[1]) : 0;
+    return sqlTypeFactory.createSqlType(SqlTypeName.DECIMAL, precision, scale);
+  }
+
+  private RelDataType getVarcharType(String type) {
+    String precision = type.substring(8, type.length() - 1);
+    return sqlTypeFactory.createSqlType(SqlTypeName.VARCHAR, Integer.parseInt(precision));
+  }
+
+  private RelDataType getCharType(String type) {
+    String precision = type.substring(5, type.length() - 1);
+    return sqlTypeFactory.createSqlType(SqlTypeName.CHAR, Integer.parseInt(precision));
+  }
+
+  private RelDataType convertType(String type) {
+    if (type.toUpperCase().startsWith("DECIMAL(")) {
+      return getDecimalType(type);
+    } else if (type.toUpperCase().startsWith("VARCHAR(")) {
+      return getVarcharType(type);
+    } else if (type.toUpperCase().startsWith("CHAR(")) {
+      return getCharType(type);
+    }
+    return sqlTypeFactory.createSqlType(SqlTypeName.valueOf(type.toUpperCase()));
+  }
+
+  @Override
+  protected SqlNode visitCast(Cast node, ParserVisitorContext context) {
+    SqlDataTypeSpec spec = SqlTypeUtil.convertTypeToSpec(convertType(node.getType()));
+    return CAST.createCall(getParserPos(node), visitNode(node.getExpression(), context), spec);
+  }
+
+  @Override
+  protected SqlNode visitFieldReference(FieldReference node, ParserVisitorContext context) {
+    return createLiteralNumber(node.getFieldIndex(), getParserPos(node));
+  }
+
+  @Override
+  protected SqlNode visitWindow(Window node, ParserVisitorContext context) {
+    SqlParserPos pos = getParserPos(node);
+    SqlNodeList partitionList = createSqlNodeList(getListSqlNode(node.getPartitionBy(), context), pos);
```

Comment:
Maybe worth introducing new method for `createSqlNodeList(getListSqlNode(x))` together.

We can rename `getListSqlNode` to `toListOfSqlNode`, and implement `createSqlNodeList(getListSqlNode(x))` in `toSqlNodeList`.